src/Arcus.Security.Providers.AzureKeyVault/Authentication/ManagedServiceIdentityAuthenticator.cs

```diff
@@ -13,6 +13,21 @@ namespace Arcus.Security.Providers.AzureKeyVault.Authentication
     public class ManagedServiceIdentityAuthenticator : IKeyVaultAuthentication, IKeyVaultAuthenticator
 #pragma warning restore 618
     {
+        private readonly string _connectionString;
+
+        private readonly string _azureAdInstance;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ManagedServiceIdentityAuthenticator"/> class.
+        /// </summary>
+        /// <param name="connectionString">The connection string to use to authenticate.</param>
+        /// <param name="azureAdInstance">The azure AD instance to use to authenticate.</param>
+        public ManagedServiceIdentityAuthenticator(string connectionString = null, string azureAdInstance = null)
```

Comment:
Frankly I think we should provide 3 overloads:
1. Constructor without any parameters
2. One with connection string only
3. One with both of these

This would allow us to guard on the values and enforce providing the required values. Next to that, it would be clearer for consumers when to use what.

Do you agree? How about you @stijnmoreels 	?

Reply:
The parameters are optional so we already have the 3 overloads you suggest.  This mirror (and slightly improves) the [`AzureServiceTokenProvider` constructor](https://github.com/Azure/azure-sdk-for-net/blob/ddda7cb74b979f03bb03e240c06c924914ee8bdd/src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs#L61)

There wouldn't be any guarding required on `connectionString`.  `null` is a valid value and causes the `AzureServiceTokenProvider` to fall back to default "magic" of looking at the `AzureServicesAuthConnectionString`, or using System Assigned Identity, VS token provider or Azure CLI token provider.

The guarding is already in place for azureAdInstance ([see comment on line 65](https://github.com/arcus-azure/arcus.security/pull/90/files/fa5e24d73361a0da9791d418622ac1390c266385#diff-e1f5781b0670280ed5307045e4c9767fR65))

Reply:
Yes and no, while I agree I think it would be good to have an empty controller next to this one at least because it might confuse people who are new to Managed Identity or do not have the technical background on how it works.

Managed Identity does "magic" for 90% of the use-cases and would prefer to keep it like that if it's ok for you?

Reply:
I agree on the other aspects concerning guarding, that's ok for me.

Reply:
OK, I'll add a parameterless constructor when I get into the office tomorrow morning and push an update.

Reply:
Done.